src/components/verification-grid-settings/verification-grid-settings.ts

```diff
@@ -48,7 +52,63 @@ export class VerificationGridSettingsComponent extends AbstractComponent(LitElem
     this.isFullscreen = document.fullscreenElement === this.verificationGrid;
   }
 
-  public render() {
+  private handleGridSizeChange(event: ChangeEvent<HTMLInputElement>) {
+    if (!this.verificationGrid) {
+      throw new Error("Could not find associated verification grid component");
+    }
+
+    // we use the Number constructor here instead of parseInt so that if
+    // the value of the input is not a number the number constructor will return
+    // NaN and we can hard fail and not change the grid size instead of
+    // changing to a grid size that the user did not expect
+    const inputValue = event.target.value;
+    const newGridSize = Number(inputValue);
+
+    // we use isFinite here to check that the value is not NaN, and that
+    // values such as Infinity are not considered as a valid grid size
+    if (!isFinite(newGridSize)) {
+      throw new Error("New grid size value could not be converted to a number");
```

Comment:
I debated this in my mind for a bit because it seems to violate the "Fail Early" principle

I agree, that this should be the responsibility of the grid component, and more specifically the attribute should check that the value provided is a valid value

Reply:
Fixed and validated through the following tests that were added to the verification grids attribute tests

![image](https://github.com/user-attachments/assets/7081f380-239a-4b1f-940f-d184fddeee75)